packages/flutter/lib/src/material/search.dart

```diff
@@ -501,51 +514,47 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
       case null:
         break;
     }
-    final String routeName;
+    String? routeName;
     switch (theme.platform) {
-      case TargetPlatform.iOS:
-      case TargetPlatform.macOS:
-        routeName = '';
-        break;
       case TargetPlatform.android:
       case TargetPlatform.fuchsia:
       case TargetPlatform.linux:
       case TargetPlatform.windows:
         routeName = searchFieldLabel;
+        break;
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
+        routeName = '';
+        break;
     }
 
     return Semantics(
       explicitChildNodes: true,
       scopesRoute: true,
       namesRoute: true,
       label: routeName,
-      child: Scaffold(
-        appBar: AppBar(
-          backgroundColor: theme.primaryColor,
-          iconTheme: theme.primaryIconTheme,
-          textTheme: theme.primaryTextTheme,
-          brightness: theme.primaryColorBrightness,
```

Comment:
These shouldn't be removed, as they are needed to set the appropriate properties on the AppBar. We could get them from the `AppBarTheme` of the theme, but that is not how the default theme is setup. We are looking at cleaning up the `AppBarTheme` soon, but for now please just put these back. This will fix the new tests that are failing.